sdk/core/core-http/test/msAssert.ts

```diff
@@ -23,7 +23,9 @@ export function throws(
   }
 
   if (!thrownError) {
-    assert.throws(() => {});
+    assert.throws(() => {
```

Comment:
what a wild piece of code.  Why wouldn't assert.throws be wrapped around syncFunction?

Reply:
If I change this to the following, it works, though the type transformation is a bit odd:

```ts
export function throws(
  syncFunction: () => void,
  expectedError?: ((error: Error) => void) | Error
): void {
  if (expectedError) {
    assert.throws(syncFunction, expectedError as any as ErrorConstructor, "Should be the expected error");
  } else {
    assert.throws(syncFunction);
  }
}
```

The types we have only have the up to three arity version, and the second parameter must be an ErrorConstructor, while Chai does allow the Error parameter.

Reply:
Hm I think this might be too aggressive of a refactor since chai doesn't seem to support taking a function that validates the expected error in this way. They seem to just do instanceof and message checks.

I think for the purposes of your PR here we should leave your original lint fix and worry about removing this method later. Sound good?

Reply:
sounds good!